arelle/Locale.py

```diff
@@ -78,7 +79,7 @@ def getLanguageCode():
     except (AttributeError, ValueError): #language code and encoding may be None if their values cannot be determined.
         return "en"
 
-def getLanguageCodes(lang=None):
+def getLanguageCodes(lang: Optional[str] = None) -> list[str, str, str]:
```

Comment:
When using add from __future__ import annotations you can type using `lang: str | None = None`, which is the Python 3.10+ way of typing.

Not sure about the list though. If it's a list of strings that's being returned, the return type is simply `list[str]`

Reply:
Yes string thing looks funny!
Didn't know about the | thing, I will fix that every where... thank you!!!

Reply:
fixed